controllers/keywordsController.js

```diff
@@ -22,4 +22,52 @@ const list = async (req, res) => {
   }
 };
 
-module.exports = { list };
+const put = async (req, res) => {
+  if (!isValidString(req.params.keywordId) || isEmptyString(req.params.keywordId)) {
+    return res.status(400).send({ message: "[InvalidKeywordId] Error occured" });
+  }
+
+  if (
+    !isValidArray(req.body.includedKeyword) ||
+    !req.body.includedKeyword.every((el) => isValidString(el))
+  ) {
+    return res.status(400).send({ message: "[InvalidIncludedKeyword] Error occured" });
+  }
+
+  if (
+    !isValidArray(req.body.excludedKeyword) ||
+    !req.body.excludedKeyword.every((el) => isValidString(el))
+  ) {
+    return res.status(400).send({ message: "[InvalidExcludedKeyword] Error occured" });
+  }
+
+  if (!isValidString(req.body.ownerUid) || isEmptyString(req.body.ownerUid)) {
+    return res.status(400).send({ message: "[InvalidOwnerUid] Error occured" });
+  }
+
+  try {
+    const { keywordId } = req.params;
+    const { includedKeyword, excludedKeyword } = req.body;
+    const keywordResult = await keywordModel
+      .findByIdAndUpdate(keywordId, { $set: { includedKeyword, excludedKeyword } }, { new: true })
+      .exec();
+    let postIdListHadExcludedKeyword = [];
+    excludedKeyword.forEach(async (keyword) => {
+      const postList = await postModel.find({ keywordId }).find({ content: { $regex: keyword } });
+      const postIdList = postList.map((post) => post._id.toString());
+      postIdListHadExcludedKeyword.push(...postIdList);
+    });
+
+    const postListResult = await postModel.find({ keywordId }).exec();
+    const postIdListResult = postListResult
+      .filter((el) => {
+        return !postIdListHadExcludedKeyword.includes(el._id.toString());
+      })
+      .map((el) => el._id);
```

Comment:
```suggestion
    const keywordResult = await postModel
    .find({ keywordId })
    .find({
      content: { $not: { $regex: excludedKeyword.join("|") }, $regex: includedKeyword.join("|") },
    });
```
저는 이 쿼리로 테스트 해봤을 때 됐었는데, 한 번 테스트 해보셔야 할 것 같아요!